### PR TITLE
[libhv] Add new port

### PR DESCRIPTION
**Pull request by ithewei**
Add a new port for https://github.com/ithewei/libhv.

- What does your PR fix? Fixes #
No associated issue.

- Which triplets are supported/not supported? Have you updated the CI baseline?
  support: x64-linux、x64-osx、x64-windows、x86-windows
  not support: arm64-windows、arm-uwp、x64-uwp

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

**Comment by ithewei** [comment]
Please review and merge! Thanks!

File: ports/libhv/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "libhv",
+  "version-string": "v1.0.0",
```
Comment (NancyLi1013) [inline ports/libhv/vcpkg.json:3]:
```suggestion
  "version": "1.0.0",
```
Please see the doc https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#22-package-versions to learn more about package versions.

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -604,6 +604,9 @@ libhdfs3:x64-windows-static=fail
 libhdfs3:x64-windows-static-md=fail
 libhdfs3:x86-windows=fail
 libhdfs3:x64-linux=fail
+libhv:arm64-windows=fail
+libhv:arm-uwp=fail
+libhv:x64-uwp=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:609]: If upstream doesn't support these triplets, please add `supports: ! (arm |  uwp) `to vcpkg.json. And remove these codes in ci.baseline.txt.

Also I would be better to add upsupported messages to portfile.cmake.
`vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/libhv/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
+vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/libhv/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")

vcpkg_from_github(
```

File: ports/libhv/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -DBUILD_EXAMPLES=OFF
+        -DBUILD_UNITTEST=OFF
+)
+
+vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp")
```
Comment (NancyLi1013) [inline ports/libhv/portfile.cmake:17]:
```suggestion
```
This line should be added to the top list of this file.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for adding this port @ithewei.

**Review by ras0219-msft — APPROVED** [review]
I made an additional change to only build one flavor at a time; LGTM and thanks for making a great library!

**Pull request merged**